BlueDBInMemory/src/main/java/io/bluedb/memory/BlueCollectionImpl.java

```diff
@@ -1,71 +1,107 @@
 package io.bluedb.memory;
 
 import java.io.Serializable;
-import java.util.Iterator;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import io.bluedb.api.BlueCollection;
 import io.bluedb.api.BlueQuery;
 import io.bluedb.api.Condition;
 import io.bluedb.api.Updater;
 import io.bluedb.api.exceptions.BlueDbException;
 import io.bluedb.api.keys.BlueKey;
+import io.bluedb.api.keys.TimeFrameKey;
+import io.bluedb.api.keys.TimeKey;
 
 class BlueCollectionImpl<T extends Serializable> implements BlueCollection<T> {
 
 	private Class<T> type;
+	private Map<BlueKey, T> data = new HashMap<>();

```

Comment:
Should we use a concurrent hash map? Or does this really matter for the in memory impl?